smoker/server/plugins/__init__.py

```diff
@@ -513,6 +510,7 @@ def run_module(self, module, **kwargs):
             raise
 
         try:
+            # TODO why are we passing self to a BasePlugin constructor?
```

Comment:
I think this was done to be able to access historical results and parameters from plugin itself.